### PR TITLE
Update skeleton plugin to support elastic extension v5

### PR DESCRIPTION
**Pull request by vishaldevgire**
Changes:
 - Implemented extension v5 calls 
 - Skeleton plugin earlier had `AgentInstances` object which was an abstraction over plugin and orchestration platform communication, with `v5` plugin plugin will be dealing with multiple clusters thats why `AgentInstances` is replaced with `Map<String, AgentInstances>` structure where multiple `AgentInstances` will be stored against `ClusterProfile` id.

### CODE REVIEW

**Comment by GaneshSPatil** [comment]
@vishaldevgire -- [getPluginSettings](https://github.com/gocd-contrib/elastic-agent-skeleton-plugin/blob/master/src/main/java/com/example/elasticagent/PluginRequest.java#L39-L48) method is not required anymore, as all the cluster profile properties are sent as part of the request.

**Comment by GaneshSPatil** [comment]
[this](https://github.com/gocd-contrib/elastic-agent-skeleton-plugin/blob/master/src/main/java/com/example/elasticagent/Constants.java#L32) constant needs to be removed.

File: src/main/java/com/example/elasticagent/requests/MigrateConfigPayload.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+    @Expose
+    private PluginSettings pluginSettings;
+
+    @Expose
+    private List<Map<String, String>> elasticAgentProfiles;
+
+    @Expose
+    private List<ClusterProfile> clusterProfiles;
```
Comment (GaneshSPatil) [inline src/main/java/com/example/elasticagent/requests/MigrateConfigPayload.java]: This won't work! as for all the other requests, we send only cluster Profile properties. For migrate config request, we sent id, plugin-id, and properties.

Refer to [this](https://plugin-api.gocd.org/19.3.0/elastic-agents/#migrate-config) documentation for more information.

Comment (vishaldevgire) [reply src/main/java/com/example/elasticagent/requests/MigrateConfigPayload.java]: Done.

**Comment by vishaldevgire** [comment]
Incorporated all code changes suggested by @GaneshSPatil

**Pull request merged**